src/lava/magma/runtime/runtime_service.py

```diff
@@ -153,6 +206,30 @@ def _relay_pm_ack_given_model_id(self, model_id: int):
         ack_relay_port = self.service_to_runtime
         ack_relay_port.send(ack_recv_port.recv())
 
+    def _handle_pause(self):
+        # Inform all ProcessModels about the PAUSE command
+        self._send_pm_cmd(MGMT_COMMAND.PAUSE)
+        rsps = self._get_pm_resp()
+        for rsp in rsps:
+            if not enum_equal(rsp,
+                              LoihiPyRuntimeService.PMResponse.STATUS_PAUSED):
+                raise ValueError(f"Wrong Response Received : {rsp}")
+        # Inform the runtime about successful pausing
+        self.service_to_runtime.send(MGMT_RESPONSE.PAUSED)
+
+    def _handle_stop(self):
```

Comment:
This can probably be handled at any time. Can't think of why we would not want to do this. If we enforce a hard stop then obviously we don't care about system state and safety.